### PR TITLE
implement isByteArrayPinned and isMutableByteArrayPinned

### PR DESCRIPTION
**Pull request by andrewthad**
Resolves https://github.com/haskell/primitive/issues/113.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Looks good to me! Can you also add an entry to the changelog?

**Comment by andrewthad** [comment]
I've really got to get changelogs integrated into my workflow better. After a lifetime of not doing them, the habit is still weak. Thanks for the reminder.

File: Data/Primitive/ByteArray.hs [hunk 1]
```diff
@@ -192,6 +200,26 @@ sizeofMutableByteArray :: MutableByteArray s -> Int
...
+#if __GLASGOW_HASKELL__ >= 802
+-- | Check whether or not the byte array is pinned. Pinned byte arrays cannot
+--   be moved by the garbage collector. It is safe to use 'byteArrayContents'
+--   on such byte arrays. This function is only available when compiling with
+--   GHC 8.2 or newer.
+isByteArrayPinned :: ByteArray -> Bool
+{-# INLINE isByteArrayPinned #-}
+isByteArrayPinned (ByteArray arr#) = case Exts.isByteArrayPinned# arr# of
+  1# -> True
+  _ -> False
```
Comment (treeowl) [inline Data/Primitive/ByteArray.hs]: This should probably use `GHC.Exts.isTrue#`

**Comment by andrewthad** [comment]
I doubt that the `INLINE` pragmas give us anything in this case. GHC will certainly decide to inline a function this small on its own. I only put them there for consistency. Every other function here has an `INLINE` pragma.

**Pull request merged**